azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -139,26 +132,28 @@ def update(self, force_update=False):
         """
         Updates the current GoalState instance fetching values from the WireServer/HostGAPlugin as needed
         """
-        xml_text, xml_doc = GoalState._fetch_goal_state(self._wire_client)
-
-        vm_settings = self._fetch_vm_settings(force_update=force_update)
+        timestamp = create_timestamp()
+        xml_text, xml_doc, incarnation = GoalState._fetch_goal_state(self._wire_client)
 
-        if force_update or self._incarnation != findtext(xml_doc, "Incarnation"):
-            # update the extended goal state, using vm_settings for the extensions (unless they are None, then use extensionsConfig)
+        if force_update or self._incarnation != incarnation:
+            # If we are fetching a new goal state
+            self._history = GoalStateHistory(timestamp, incarnation)
             self._initialize_basic_properties(xml_doc)
-            self._fetch_extended_goal_state(xml_text, xml_doc,  vm_settings)
+            self._fetch_extended_goal_state(xml_text, xml_doc, force_vm_settings_update=force_update)
         else:
-            # else just ensure the extensions are using the latest vm_settings
-            if vm_settings is not None:
+            # else ensure the extensions are using the latest vm_settings
+            timestamp = create_timestamp()
```

Comment:
why do we need different timestamp? can't we use L#135

Reply:
L#136 calls _fetch_goal_state, which does a network call, which can have delays. we need a new timestamp here